Source/NSLayoutConstraint.m

```diff
@@ -0,0 +1,589 @@
+      relation = @"=";
+      break;
+    case NSLayoutRelationGreaterThanOrEqual:
+      relation = @">=";
+      break;
+    default:
+      break;
+    }
+
+  return relation;
+}
+
++ (NSLayoutRelation) _stringToRelation: (NSString *)str
+{
+  NSLayoutRelation r = 0;
+
+  if ([@"<=" isEqualToString: str])
+    {
+      r = NSLayoutRelationLessThanOrEqual;
+    }
+  else if ([@"=" isEqualToString: str])
+    {
+      r = NSLayoutRelationEqual;
+    }
+  else if ([@">=" isEqualToString: str])
+    {
+      r = NSLayoutRelationGreaterThanOrEqual;
+    }
+  
+  return r;
+}
+
++ (void) _activateConstraint: (NSLayoutConstraint *)constraint
+{
+  [activeConstraints addObject: constraint];
+  activeConstraints = [[activeConstraints sortedArrayUsingSelector: @selector(compare:)] mutableCopy];
+}
+
++ (void) _removeConstraint: (NSLayoutConstraint *)constraint
+{
+  [activeConstraints removeObject: constraint];
+}
+
++ (NSArray *) constraintsWithVisualFormat: (NSString *)fmt 
+                                  options: (NSLayoutFormatOptions)opt 
+                                  metrics: (NSDictionary *)metrics 
+                                    views: (NSDictionary *)views
+{
+  NSMutableArray *array = [NSMutableArray arrayWithCapacity: 10];
+  return array;
+}
+
+- (instancetype) initWithItem: (id)firstItem 
+                    attribute: (NSLayoutAttribute)firstAttribute 
+                    relatedBy: (NSLayoutRelation)relation 
+                       toItem: (id)secondItem
+                    attribute: (NSLayoutAttribute)secondAttribute 
+                   multiplier: (CGFloat)multiplier
+                     constant: (CGFloat)constant
+                     priority: (CGFloat)priority;
+{
+  self = [super init];
+  if (self != nil)
+    { 
+      _firstItem = firstItem;
+      _secondItem = secondItem;
+      _firstAttribute = firstAttribute;
+      _secondAttribute = secondAttribute;
+      _relation = relation;
+      _multiplier = multiplier;
+      _constant = constant;
+      _priority = priority;
+      
+      [NSLayoutConstraint _activateConstraint: self];
+    }
+  return self;
+}
+
++ (instancetype) constraintWithItem: (id)view1 
+                          attribute: (NSLayoutAttribute)attr1 
+                          relatedBy: (NSLayoutRelation)relation 
+                             toItem: (id)view2 
+                          attribute: (NSLayoutAttribute)attr2 
+                         multiplier: (CGFloat)mult 
+                           constant: (CGFloat)c
+{
+  NSLayoutConstraint *constraint =
+    [[NSLayoutConstraint alloc] initWithItem: view1
+                                   attribute: attr1
+                                   relatedBy: relation
+                                      toItem: view2
+                                   attribute: attr2
+                                  multiplier: mult
+                                    constant: c
+                                    priority: 1000.0];
```

Comment:
Why didn't you use the constant NSLayoutPriorityRequired here?

Reply:
Done